knora/dsplib/utils/xml_upload.py

```diff
@@ -603,21 +611,42 @@ def xml_upload(input_file: str, server: str, user: str, password: str, imgdir: s
 
     res_iri_lookup: Dict[str, str] = {}
 
+    failed_uploads = []
     for resource in resources:
-        if verbose:
-            resource.print()
-        if resource.bitstream:
-            img = sipi.upload_bitstream(os.path.join(imgdir, resource.bitstream))
-            bitstream = img['uploadedFiles'][0]['internalFilename']
-        else:
-            bitstream = None
-
-        # create the resource on the server
-        instance: ResourceInstance = res_classes[resource.restype](con=con, label=resource.label,
-                                                                   permissions=permissions_lookup.get(
-                                                                       resource.permissions),
-                                                                   bitstream=bitstream,
-                                                                   values=resource.get_propvals(res_iri_lookup,
-                                                                                                permissions_lookup)).create()
-        res_iri_lookup[resource.id] = instance.iri
-        print("Created resource:", instance.label, "(", resource.id, ") with IRI", instance.iri)
+        bitstream = None
+        try:
+            if verbose:
+                resource.print()
+            if resource.bitstream:
+                img = sipi.upload_bitstream(os.path.join(imgdir, resource.bitstream))
+                bitstream = img['uploadedFiles'][0]['internalFilename']
+
+            # create the resource on the server
+            instance = res_classes[resource.restype](con=con, label=resource.label,
+                                                     permissions=permissions_lookup.get(resource.permissions),
+                                                     bitstream=bitstream,
+                                                     values=resource.get_propvals(res_iri_lookup,
+                                                                                  permissions_lookup)).create()
+            res_iri_lookup[resource.id] = instance.iri
+            print(f"Created resource '{instance.label}' ({resource.id}) with IRI '{instance.iri}'")
+
+        except BaseError as err:
+            failed_uploads.append(resource.id)
+            print(f"ERROR while trying to upload '{resource.label}' ({resource.id}). The error message was: {err.message}")
+
+        except Exception as exception:
+            failed_uploads.append(resource.id)
+            print(f"ERROR while trying to upload '{resource.label}' ({resource.id}). The error message was: {str(exception)}")
```

Comment:
```suggestion
            print(f"ERROR while trying to upload '{resource.label}' ({resource.id}). The error message was: {exception}")
```

f-strings automatically call str() on any non-string object